src/libstd/time.rs

```diff
@@ -357,6 +357,14 @@ impl SystemTime {
     pub fn elapsed(&self) -> Result<Duration, SystemTimeError> {
         SystemTime::now().duration_since(*self)
     }
+
+    /// Returns `Some(t)` where `t` is the time `self + duration` if `t` can be represented as
+    /// `SystemTime` (which means it's inside the bounds of the underlying data structure), `None`
+    /// otherwise.
+    #[stable(feature = "time_checked_add", since = "1.32.0")]
```

Comment:
This should be added as unstable initially.

I think we'd also want to call it `checked_add` and have it take the `Duration` by value to match with conventions elsewhere.

Reply:
> I think we'd also want to call it `checked_add` and have it take the `Duration` by value to match with conventions elsewhere.

I guess I should only rename the public function. Do you think I should make all internal `checked_add_duration` functions take the `Duration` argument by value or only the public facing one?

Reply:
The internal function naming doesn't matter too much IMO - I think it's fine as-is since otherwise you'd just have free functions named `add` which seems a bit iffy.